myokit/_sim/cmodel.py

```diff
@@ -0,0 +1,428 @@
+
+        # Create output lists
+        dependents = []
+        independents = []
+
+        # Check dependents, make sure all are Name or Derivative objects from
+        # the cloned model.
+        for x in deps:
+            deriv = False
+            if isinstance(x, myokit.Variable):
+                var = model.get(x.qname())
+            elif isinstance(x, myokit.Name):
+                var = model.get(x.var().qname())
+            elif isinstance(x, myokit.Derivative):
+                deriv = True
+                var = model.get(x.var().qname())
+            else:
+                x = str(x)
+                if x[:4] == 'dot(' and x[-1:] == ')':
+                    deriv = True
+                    var = x = x[4:-1]
+                var = model.get(x)
+            lhs = myokit.Name(var)
+            if deriv:
+                lhs = myokit.Derivative(lhs)
+                if not var.is_state():
+                    raise ValueError(
+                        'Sensitivity of ' + lhs.code() + ' requested, but '
+                        + var.qname() + ' is not a state variable.')
+            elif var.is_bound():
+                raise ValueError(
+                    'Sensitivities cannot be calculated for bound'
+                    ' variables (got ' + str(var.qname()) + ').')
+            # Note: constants are fine, just not very useful! But may be
+            # easy, e.g. when working with multiple models.
+            dependents.append(lhs)
+
+        # Check independents, make sure all are Name or InitialValue
+        # objects from the cloned model.
+        for x in indeps:
+            init = False
+            if isinstance(x, myokit.Variable):
+                var = self.model.get(x.qname())
+            elif isinstance(x, myokit.Name):
+                var = self.model.get(x.var().qname())
+            elif isinstance(x, myokit.InitialValue):
+                init = True
+                var = model.get(x.var().qname())
+            else:
+                x = str(x)
+                if x[:5] == 'init(' and x[-1:] == ')':
+                    init = True
+                    x = x[5:-1]
+                var = model.get(x)
+            lhs = myokit.Name(var)
+            if init:
+                lhs = myokit.InitialValue(myokit.Name(var))
+                if not var.is_state():
+                    raise ValueError(
+                        'Sensitivity with respect to ' + lhs.code() +
+                        ' requested, but ' + var.qname() + ' is not a'
+                        ' state variable.')
+            elif not var.is_literal():
+                raise ValueError(
+                    'Sensitivity with respect to ' + var.qname() +
+                    ' requested, but this is not a literal variable (it'
+                    ' depends on other variables).')
+            independents.append(lhs)
+
+        return True, dependents, independents
+
+    def _derive_sensitivity_equations(self, model, deps, indeps):
```

Comment:
This function seems to always return empty lists. Is it possible that in line 256 it was supposed to be 
```
output_variables = [
            lhs.var() for lhs in deps
            if (isinstance(lhs, myokit.Name) and lhs.var().is_state())]
```

Reply:
Ahh I just read the comment above, so you intentionally ignore states?

Reply:
Yeah, the d/dp of the state variables is already known at this point, but in cardiac models we usually have a lot of variables of interest that aren't state variables, but some function of them instead. So this method is meant to find the equations to calculate the sensitivities of those "intermediate variables"